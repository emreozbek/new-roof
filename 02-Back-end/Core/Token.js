import jwt from "jsonwebtoken";
import { KEY } from "../Config";

class Token {
  constructor() {
    this.whiteList = ["/User/Login"];
    this.KEY = KEY;
  }

  Create(data) {
    return jwt.sign(data, this.KEY, { expiresIn: 4000 });
  }
  CheckWhiteList(url) {
    if (this.whiteList.indexOf(url) === -1) return false;
    return true;
  }
  Verify(token) {
    let res = true;
    try {
      jwt.verify(token, this.KEY);
    } catch (e) {
      res = false;
    }
    return res;
  }
  Decode(token) {
    return jwt.decode(token, this.KEY);
  }
}

export default Token;
