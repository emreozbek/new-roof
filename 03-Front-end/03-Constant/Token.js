const SET_TOKEN = "SET_TOKEN";

export { SET_TOKEN };
