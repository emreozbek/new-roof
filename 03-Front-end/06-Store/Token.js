const Token = {
  user: {},
  token: ""
};

export default Token;
