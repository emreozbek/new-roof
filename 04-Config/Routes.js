import AppConfig from "./App";

const domain = `${AppConfig.host}:${AppConfig.server.port}`;
const Routes = {
  user: {
    login: `${domain}/User/Login`
  }
};

module.exports = Routes;
