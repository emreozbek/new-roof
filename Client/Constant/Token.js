export const SET_TOKEN = "SET_TOKEN";
export const DESTROY_TOKEN = "DESTROY_TOKEN";
