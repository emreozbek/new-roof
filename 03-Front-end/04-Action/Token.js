import * as Token from "../03-Constant/Token";

const setToken = data => dispatch =>
  dispatch({
    type: Token.SET_TOKEN,
    data
  });

export { setToken };
