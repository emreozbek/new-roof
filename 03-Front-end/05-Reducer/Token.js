import * as actions from "../03-Constant/Token";
import store from "../06-Store/Token";

const Token = (state = store, action) => {
  switch (action.type) {
    case actions.SET_TOKEN: {
      return {
        ...state,
        user: action.data.user,
        token: action.data.token
      };
    }
    default:
      return state;
  }
};

export default Token;
