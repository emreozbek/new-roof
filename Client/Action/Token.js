import * as Token from "../Constant/Token";

const SetToken = data => dispatch => {
  sessionStorage.user = JSON.stringify(data);
  return dispatch({
    type: Token.SET_TOKEN,
    data
  });
};

const DestroyToken = () => dispatch => {
  const data = {
    token: ""
  };
  sessionStorage.user = JSON.stringify(data);
  return dispatch({
    type: Token.DESTROY_TOKEN,
    data
  });
};

export { SetToken, DestroyToken };
