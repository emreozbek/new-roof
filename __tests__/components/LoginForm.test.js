import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import LoginForm from "../../Client/Component/LoginForm/Index";
import * as actions from "../../Client/Action/Token";
import Routes from "../../Config/Routes";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const loginForm = shallow(
    <LoginForm actions={actions} url={Routes.user.login} />
  );
  expect(loginForm).toMatchSnapshot();
});
