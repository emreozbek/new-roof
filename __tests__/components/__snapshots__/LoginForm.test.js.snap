// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Login
  actions={
    Object {
      "DestroyToken": [Function],
      "SetToken": [Function],
    }
  }
  classes={
    Object {
      "actions": "Login-actions-4",
      "button": "Login-button-2",
      "card": "Login-card-3",
      "copyright": "Login-copyright-6",
      "processing": "Login-processing-5",
      "progress": "Login-progress-1",
    }
  }
  url="http://localhost:4000/User/Login"
/>
`;
